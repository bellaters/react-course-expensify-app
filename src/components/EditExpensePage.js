import React from 'react';

const EditExpensePage = (props) => {
    console.log(props);
    return (
        <div>
        Editing the expensive with id of {props.match.params.id}
        </div>
    );
};
export default EditExpensePage;