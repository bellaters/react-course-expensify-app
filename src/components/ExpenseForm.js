import React from 'react';

export default class ExpenseForm extends React.Component {
    state = {
        description: '',
        note: ''
    };
    // CHALLENGE 105
    // setup note state 
    // setup onChange and value for textarea

    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState( () => ({ note }));
    };
    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState( () => ({ description }));
    };
    render() {
        return (
            <div>
                <form>
                    <input 
                        type="text"
                        placeholder="Description"
                        autoFocus
                        value={this.state.description}
                        onChange={this.onDescriptionChange}
                    />
                    <input 
                        type="number"
                        placeholder="Amount"
                    />
                    <textarea
                        placeholder="Add a note for your expense (optional)"
                        value={this.state.note}
                        onChange={this.onNoteChange}
                    >
                    </textarea>
                    <button>Add Expense</button>
                </form>
            </div>
        )
    }
}