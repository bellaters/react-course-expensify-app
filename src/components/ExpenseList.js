import React from 'react';
import { connect } from 'react-redux';

const ExpenseList = (props) => (
    <div>
        <h1>Expense List</h1>
        <p>
            {props.expenses.length}
        </p>
    </div>
);

const ConnectedExpenseList = connect((state) => {
    return {
        expenses: state.expenses
    };
})(ExpenseList);

export default ConnectedExpenseList;