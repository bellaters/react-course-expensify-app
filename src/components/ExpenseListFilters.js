import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAmount } from '../actions/filters';


const ExpenseListFilters = (props) => (
    <div>
        <input 
            type="text" 
            value={props.filters.text}
            onChange={(e)=>{
                props.dispatch(setTextFilter(e.target.value));
        }}/>
        <select 
            value={props.filters.sortBy}
            onChange={(e)=>{
                const sortValue = e.target.value;
                switch(sortValue){
                    case 'date':
                        props.dispatch(sortByDate());
                    case 'amount':
                        props.dispatch(sortByAmount());
                }
                console.log(sortValue);
        }}>
            <option value="date">Date</option>
            <option value="amount">Amount</option>
        </select>
    </div>
);

// CHALLENGE 104.1
// setup value and onChange for select

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
};

export default connect(mapStateToProps)(ExpenseListFilters);