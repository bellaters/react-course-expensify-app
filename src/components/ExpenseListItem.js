import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

export const ExpenseListItem = ({ id, description, amount, createAt}) => (
    <Link to={`/edit/${id}`}>
        <div>
            <h3>{description}</h3>
            <span>{moment(createAt).format('MMMM DD, YYYY')}</span>
        </div>
            <h3>{numeral(amount / 100).format('$0,0.00')}</h3>
    </Link>
);

export default ExpenseListItem;
