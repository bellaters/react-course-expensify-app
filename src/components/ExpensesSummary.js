// 1. Create ExpensesSummary Component
// -----------------------------------
import React from 'react';
import { connect } from 'react-redux';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => (
    <div>
        <p>
        {`Viewing ${props.expenses.length} expenses 
        totalling ${selectExpensesTotal(props.expenses)}`}
        </p>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpensesTotal(state.expenses, state.filters)
    };
}; 

export default connect(mapStateToProps)(ExpensesSummary);

// Rendered by ExpenseDashboardPage
// Test with two snapshot tests
// Connected to store for:
//      - expenseCount (how many visible expenses?)
//      - expensesTotal (what's the total of the visible expenses?)

// Example
// Viewing 2 expenses totalling $94.34

// Example
// Viewing 3 expenses totalling $94.34

// 2. Commit and Deploy
// ------------------------------------

// Get the feature live!