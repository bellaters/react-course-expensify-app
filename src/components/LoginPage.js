
import React from 'react';

// 1. Create LoginPage component with 'Login' button
// 2. Add snapshot test for LoginPage
// 3. Show Login component at root of app -> /
// 4. Show ExpenseDashboardPage at -> /dashboard
export const LoginPage = (props) => (
    <div>
        <button>
            Login
        </button>        
    </div>
);

export default LoginPage;
