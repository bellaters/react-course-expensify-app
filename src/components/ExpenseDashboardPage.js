import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';
//import Header from './Header';

export const ExpenseDashboardPage = () => (
    <div>
        <ExpensesSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);
export default ExpenseDashboardPage;
