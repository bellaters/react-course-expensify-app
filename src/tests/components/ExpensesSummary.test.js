import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';


test(' should correctly rendere ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary count={1} total={100} />);
    expect(wrapper).toMatchSnapshot();
});

test(' should correctly rendere ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary count={10} total={1000} />);
    expect(wrapper).toMatchSnapshot();
});