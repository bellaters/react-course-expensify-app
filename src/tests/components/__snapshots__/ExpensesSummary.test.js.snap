// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` should correctly rendere ExpensesSummary with 1 expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling 
    $1.00
     .
  </h2>
</div>
`;

exports[` should correctly rendere ExpensesSummary with multiple expenses 1`] = `
<div>
  <h2>
    Viewing 
    10
     
    expenses
     totalling 
    $10.00
     .
  </h2>
</div>
`;
